### PR TITLE
Add a simple key-value test

### PR DESCRIPTION
**Pull request by rylev**
Before we merge this, there's one open question:

The test has an assumption that a key-value store labeled "default" exists. In Spin CLI, this is the case without any further action, but for other runtimes, there might not be a "default" key-value store by default, and the test runner might need to take action to create the store. 

We'll need to figure out some mechanism for the test to specify that it is relying on the store existing.

### CODE REVIEW

**Comment by lann** [comment]
> We'll need to figure out some mechanism for the test to specify that it is relying on the store existing.

Would it make sense to have some kind of explicit "precondition" asserts that document this kind of requirement?

File: components/key-value/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Key Value
+
+Tests the key/value interface.
+
+## Expectations
+
+This test component expects the following to be true:
+* It is given permission to open a connection to a store named "default".
+* It does not have permission to access a store named "forbidden".
+* It is empty
```
Comment (lann) [inline components/key-value/README.md:10]: What is empty? `default`?

Comment (rylev) [reply components/key-value/README.md:10]: Yes, I'll make that clear.

**Comment by rylev** [comment]
> Would it make sense to have some kind of explicit "precondition" asserts that document this kind of requirement?

Yes, I'm going to try to come up with something like this. We'll likely have to change it over time, but hopefully we'll eventually land on something that works.

**Comment by rylev** [comment]
Ok I've changed the "services" config to a general "preconditions" config which covers both the services that need to run as well as other preconditions (e.g., which stores need to exist).

**Pull request merged**